src/perception_sensor_lib/src/landmark_filter/consecutive_counter_filter.cpp

```diff
@@ -0,0 +1,68 @@
+#include "perception_sensor_lib/landmark_filter/consecutive_counter_filter.hpp"
+
+#include <iostream>
+Eigen::VectorXd ConsecutiveCounterFilter::filter(const Eigen::VectorXd& state,
+                                                 const Eigen::MatrixXd& covariance,
+                                                 const Eigen::VectorXd& observations,
+                                                 const Eigen::VectorXd& observation_confidences,
+                                                 const Eigen::VectorXi& associations) {
+  // Get only the observations that are considered new relative to the SLAM map
+  Eigen::VectorXd new_observations;
+  Eigen::VectorXd new_confidences;
+  int number_of_observations = associations.size();
+  for (int i = 0; i < number_of_observations; i++) {
+    if (associations(i) == -1) {
+      new_observations.conservativeResize(new_observations.size() + 2);
+      new_observations.segment(new_observations.size() - 2, 2) = observations.segment(i * 2, 2);
+      new_confidences.conservativeResize(new_confidences.size() + 1);
+      new_confidences(new_confidences.size() - 1) = observation_confidences(i);
+    }
+  }
+  // Associate the observations that are considered new with the map stored in this filter
+  this->map.segment(0, 3) = state.segment(0, 3);
+  Eigen::VectorXi new_associations = this->_data_association_->associate(
```

Comment:
There is something that is causing me a lot of confusion: aren't you making a mistake here? You are calling this a map and sending it to the data association as if it had coordinates in a static frame. However, what is being fed into the map are observation coordinates. I will need you to explain how this works exactly